### DISCUSSION TITLE
Issue with javafx css

### DISCUSSION THREAD

**Discussion by LilRonin** [GSI-CS-CO/chart-fx#526] [Q&A] [answered]
Hello, i've got an exception with my code ony when i try using zoom class, i don't really understand where it come from since i can't find javafx css module on maven repository 
`Caused by: java.util.ServiceConfigurationError: org.controlsfx.glyphfont.GlyphFont: Provider org.controlsfx.glyphfont.FontAwesome could not be instantiated`
`Unable to make public static com.sun.javafx.css.StyleManager com.sun.javafx.css.StyleManager.getInstance() accessible: module javafx.graphics does not "exports com.sun.javafx.css" to unnamed module `
What should i add?

I'm on javafx graphics 19 i'm currently searching for javafx.css.stylemanager but i don't find it is this the issue?
https://javadoc.io/doc/org.openjfx/javafx-graphics/latest/javafx.graphics/javafx/css/package-summary.html

**Comment by wirew0rm** [comment, marked as answer]
Hey, the `com.sun.javafx.css` package is not exported and as such also not part of the public documentation of openjdk. It is still contained in the last version of [openjfx](https://github.com/openjdk/jfx/blob/19%2B8/modules/javafx.graphics/src/main/java/com/sun/javafx/css/StyleManager.java). You need to provide the appropriate command line flags to make this part of the API accessible. Since this dependency is introduced by controlsfx, you can find a lot of issues there with different solutions depending on the build system you are using. e.g. #161, controlsfx/controlsfx#1245 and [controlsfx wiki](https://github.com/controlsfx/controlsfx/wiki/Using-ControlsFX-with-JDK-9-and-above#glyphfont).

**Reply by LilRonin** [reply]
Hello, thx a lot, according to your links i had to add this to my run configuration 
`--add-exports=javafx.graphics/com.sun.javafx.css=ALL-UNNAMED`
i don't really understand what is it and how it works but i'll search for more.

**Reply by wirew0rm** [reply]
Glad it worked for you! If you want to dive in deeper, the relevant keywords are "Java Platform Module System" (JPMS) and "Jigsaw". Basically it introduces a `module.info` file which specifies which packages are public and which are internal. It was introduced by modularization of JDK9, but since the migration is really difficult in many cases, many projects have not migrated and this leads to the necessity to specify additional access rules on the command line. For the cli options https://nipafx.dev/five-command-line-options-hack-java-module-system/ gives a good introduction.
